plugins/dbviewer/api/api.js

```diff
@@ -310,25 +313,70 @@ var common = require('../../../api/utils/common.js'),
                 return callback(false);
             }
         }
+
         /**
         * Get aggregated result by the parameter on the url
         * @param {string} collection - collection will be applied related query
         * @param {object} aggregation - aggregation object
         * */
         function aggregate(collection, aggregation) {
-            var skip = parseInt(params.qstring.iDisplayStart || 0);
-            if (skip) {
-                aggregation.push({"$skip": skip});
-            }
-            if (params.qstring.iDisplayLength) {
-                aggregation.push({"$limit": parseInt(params.qstring.iDisplayLength)});
-            }
-            dbs[dbNameOnParam].collection(collection).aggregate(aggregation, function(aggregationErr, result) {
-                if (!aggregationErr) {
-                    common.returnOutput(params, {sEcho: params.qstring.sEcho, iTotalRecords: result.length, iTotalDisplayRecords: result.length, "aaData": result});
+            aggregation.push({"$count": "total"});
+            dbs[dbNameOnParam].collection(collection).aggregate(aggregation, function(err, total) {
```

Comment:
Here, we would already have a problem, if this count takes more than HTTP timeout.
So maybe let's just get rid of count and execute correct query right away, without counting the result